### PR TITLE
perf: do not trigger change detection on all keydown events

### PR DESCRIPTION
**Pull request by arturovt**

### CODE REVIEW

File: projects/angular-calendar/src/modules/common/keydown-enter.directive.ts [hunk 1]
```diff
@@ -1,17 +1,46 @@
...
+  OnDestroy,
+} from '@angular/core';
 
 @Directive({
   selector: '[mwlKeydownEnter]',
 })
-export class KeydownEnterDirective {
-  @Output('mwlKeydownEnter') keydown = new EventEmitter<any>(); // tslint:disable-line
+export class KeydownEnterDirective implements OnDestroy {
+  @Output('mwlKeydownEnter') keydown = new EventEmitter<KeyboardEvent>(); // tslint:disable-line
```
Comment (mattlewis92) [inline projects/angular-calendar/src/modules/common/keydown-enter.directive.ts:15]: The any type here is deliberate, as adding this breaks compatibility with angular 6 and 7 which are still supported (https://github.com/mattlewis92/angular-calendar/commit/d64138fe1684b9ede34af7d15d978a98166cc6ce)

File: projects/angular-calendar/src/modules/common/keydown-enter.directive.ts [hunk 2]
```diff
@@ -1,17 +1,46 @@
...
 
-  @HostListener('keydown', ['$event'])
-  onKeyPress(event: any) {
-    if (event.keyCode === 13 || event.which === 13 || event.key === 'Enter') {
-      event.preventDefault();
```
Comment (mattlewis92) [inline projects/angular-calendar/src/modules/common/keydown-enter.directive.ts:12]: I think we need to keep the preventDefault and stopPropagation?

File: projects/angular-calendar/src/modules/common/keydown-enter.directive.ts [hunk 3]
```diff
@@ -1,17 +1,46 @@
...
-      this.keydown.emit(event);
-    }
+  private keydownListener: VoidFunction;
+
+  constructor(
+    host: ElementRef<HTMLElement>,
+    ngZone: NgZone,
+    renderer: Renderer2
+  ) {
+    ngZone.runOutsideAngular(() => {
```
Comment (mattlewis92) [inline projects/angular-calendar/src/modules/common/keydown-enter.directive.ts:24]: Can we run this inside of the `ngOnInit` hook?

**Review by mattlewis92 — CHANGES_REQUESTED** [review]
Nice one, thank Artur! Left a couple of small comments 😄

File: projects/angular-calendar/src/modules/common/keydown-enter.directive.ts (hunk 1 above)
Comment (arturovt) [reply projects/angular-calendar/src/modules/common/keydown-enter.directive.ts]: Done

File: projects/angular-calendar/src/modules/common/keydown-enter.directive.ts (hunk 2 above)
Comment (arturovt) [reply projects/angular-calendar/src/modules/common/keydown-enter.directive.ts:12]: Done

File: projects/angular-calendar/src/modules/common/keydown-enter.directive.ts (hunk 3 above)
Comment (arturovt) [reply projects/angular-calendar/src/modules/common/keydown-enter.directive.ts]: Done

**Review by mattlewis92 — APPROVED** [review]
LGTM!

**Pull request merged**